src/http-client.ts

```diff
@@ -67,10 +74,15 @@ export class HttpClient {
     const logger = opts.logger || console;
     const start = new Date().getTime();
 
+    opts.customHeaders = {
+      ...this.customHeaders,
+      ...opts.customHeaders
+    };
```

Comment:
Oh this needs a change. I meant for this to include `|| {}` in the case of either of these being undefined or something like that.

Otherwise the thinking here (which Jon and I worked through) is that though we can't think of a reason why this would be the case, but if there is a scenario where we need to pass in a different value for `customHeaders` than what is passed in for the Streaming Client config, then we should use that passed in value _over_ the config value which is hopefully what will happen here.  Again, we can't think of a reason this would happen, but if someone passes in a different `customHeaders`, there would be a _reason_ behind it so it needs to be honored.